src/main/java/me/karakelley/http/controllers/Router.java

```diff
@@ -16,6 +17,9 @@ public Router route(String path, Controller controller) {
 
   @Override
   public Response respond(Request request) {
+    if (PublicDirectory.validResource(request.getPath())) {
+      return mappings.getOrDefault(request.getPath(), new StaticFilesController()).respond(request);
```

Comment:
The structure of this method seems a little strange to me, I think because you're checking which "fallback" controller to use when you may not need to use either fallback (if there is an explicit mapping for the request path). There's probably a way to make this read a little more descriptively in terms of "route priorities" (for lack of a better term). (Might require abandoning `getOrDefault`.)

Reply:
Also, beyond just code clarity, it's probably good to avoid the I/O hit of checking whether something is a valid resource if you can.

Reply:
Can you expand on this? Is there a way validate and avoid hitting I/O altogether? At some point you have to check the file system to verify it exists, right?

Reply:
Ah, yeah I was unclear. If the path is in your predefined route mappings, you don't have to check the filesystem. For example, the `GET /redirectme` should probably do its redirection without having to look at the filesystem (at least not until it handles the redirected-to target, `/`).